client/my-sites/importer/site-importer/url-validation.js

```diff
@@ -0,0 +1,45 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+import url from 'url';
+import { isWebUri } from 'valid-url';
+import { translate } from 'i18n-calypso';
+import { inRange, trim } from 'lodash';
+
+const parseUrl = function( value = '' ) {
+	const parsedUrl = url.parse( value, false, true );
+
+	// Ensure parsed url has host/hostname by prefixing with protocol, if needed.
+	return parsedUrl.protocol ? parsedUrl : url.parse( 'https://' + value );
+};
+
+const isValidUrl = function( parsedUrl ) {
+	const { hostname } = parsedUrl;
+
+	// `isWebUri` considers `http://a` valid, so check for a top level domain name as well.
+	const hasTld = inRange( hostname.lastIndexOf( '.' ), 1, hostname.length - 2 );
+
+	return isWebUri( url.format( parsedUrl ) ) && hasTld;
+};
+
+export function validateImportUrl( value ) {
```

Comment:
Shall we cut out the original source of this validation from `site-importer-input-pane.jsx` and put this utility to work there as part of this PR? Or shall we save that for a follow up? I'd be happy with either :)

Reply:
I vote for any changes to Settings > Import to happen in a follow-up to this PR, as I have a sense there may be some other clean-up and improvements that can happen at the same time.

Reply:
I went ahead and updated so that this is used in `client/my-sites/importer/site-importer/site-importer-input-pane.jsx` as well.